Fix/vite preprocess import (Security), update colors to Vermilion, trivial fixes

jdorfman opened the pull request:
Before:
<img width="1368" height="678" alt="CleanShot 2025-12-19 at 09 31 35@2x" src="https://github.com/user-attachments/assets/2d231a6e-c431-4e32-9d52-12bf9121529a" />


After:
<img width="1027" height="499" alt="image" src="https://github.com/user-attachments/assets/33f986d7-9e69-4a9b-a0b0-bb76357bd802" />

jdorfman committed:
Fix vitePreprocess import path in svelte.config.js

- Change import from '@sveltejs/kit/vite' to '@sveltejs/vite-plugin-svelte'
- Resolves SyntaxError: module does not provide export named 'vitePreprocess'

Amp-Thread-ID: https://ampcode.com/threads/T-c88bfafb-d4c0-454f-a2b1-b429beda9af5
Co-authored-by: Amp <amp@ampcode.com>

jdorfman committed:
chore: update dependencies in package.json

- Upgrade @sveltejs/adapter-static from ^2.0.1 to ^3.0.10
- Upgrade @sveltejs/kit from ^1.5.0 to ^2.49.2
- Upgrade svelte from ^3.54.0 to ^5.45.8
- Upgrade vite from ^4.0.0 to ^6.4.1
- Upgrade vitest from ^0.31.0 to ^4.0.15

jdorfman committed:
style: update color variables in app.css

- Change primary and secondary color values to new vermilion shades
- Add new vermilion color variants for enhanced theming

jdorfman committed:
fix: correct spelling of "GitHub" in filters.js and searchExamples.yaml

- Update "Github" to "GitHub" for consistency across the codebase
- Ensure proper naming in search examples for better clarity

jdorfman committed:
style: enhance color variables and button styles in app.css and tailwind.config.cjs

- Add new color variables for primary hover and dark text
- Update button styles for primary and secondary buttons to improve hover effects and text colors
- Ensure consistent theming across light and dark modes

jdorfman committed:
chore: update Node.js version in GitHub Actions workflow

- Change Node.js version from 16 to 20 in the test workflow
- Ensures compatibility with the latest features and improvements

jdorfman committed:
Update GitHub Actions to v4 and Node.js to 22

jdorfman committed:
chore: update .gitignore to exclude local settings file

- Add .claude/settings.local.json to .gitignore to prevent local configuration from being tracked

jdorfman committed:
chore: downgrade Node.js version in GitHub Actions workflow

- Change Node.js version from 22 to 18.13 in the test workflow
- Ensures compatibility with existing project dependencies

jdorfman committed:
chore: upgrade Node.js version in GitHub Actions workflow

- Change Node.js version from 18.13 to 22.0.0 in the test workflow
- Ensures compatibility with the latest features and improvements

trly commented on .github/workflows/test.yml:
```diff
@@ -12,12 +12,12 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
```
I'd move this to the latest version of checkout, just since we don't touch this that often
https://github.com/actions/checkout/releases/tag/v6.0.1

jdorfman replied:
Good call. Updated.

trly commented on .github/workflows/test.yml:
```diff
@@ -12,12 +12,12 @@ jobs:
+    - uses: actions/checkout@v4
 
     - name: Setup Node.js
-      uses: actions/setup-node@v3
```
same with setup-node
https://github.com/actions/setup-node/releases/tag/v6.1.0

jdorfman replied:
Another good call. Updated.

trly reviewed (changes requested):
Other than the GH action versions, everything else looks 👍🏻

jdorfman committed:
chore: upgrade GitHub Actions to v6 in test workflow

- Update actions/checkout from v4 to v6
- Update actions/setup-node from v4 to v6
- Ensures compatibility with the latest GitHub Actions features
